src/shared/filesystem.ts

```diff
@@ -18,7 +20,52 @@ export interface Stats extends fs.Stats {
 // functions
 export const access = promisify(fs.access)
 
-export const mkdir = promisify(fs.mkdir)
+const _mkdir = promisify(fs.mkdir)
+interface ErrorWithCode {
+    code?: string
+}
+
+export async function mkdir(
+    path: PathLike,
+    options?: number | string | MakeDirectoryOptions | undefined | null,
+): Promise<void> {
+    try {
+        await _mkdir(path, options)
+    } catch (err) {
+        // mkdir calls with recurse do not work as expected when called through electron.
+        // See: https://github.com/nodejs/node/issues/24698#issuecomment-486405542
+        const error = err as ErrorWithCode
+        if (error.code && error.code === 'ENOENT') {
+            if (options && typeof options === 'object' && options.recursive && typeof path === 'string') {
+                await mkdirRecursive(path, options)
+
+                return
+            }
+        }
+
+        throw err
+    }
+}
+
+async function mkdirRecursive(
+    path: string,
+    options: MakeDirectoryOptions,
+): Promise<void> {
+    try {
+        await access(path)
+
+        // path exists, nothing left to do
+        return
```

Comment:
The return statement is redundant, as there is no code following it.

Reply:
Thanks - I rearranged the code and didn't catch that before PR